src/Avalonia.Base/Platform/PixelFormat.cs

```diff
@@ -1,9 +1,99 @@
-﻿namespace Avalonia.Platform
+﻿using System;
+
+namespace Avalonia.Platform
 {
-    public enum PixelFormat
+    internal enum PixelFormatEnum
     {
         Rgb565,
         Rgba8888,
-        Bgra8888
+        Bgra8888,
+        BlackWhite,
+        Gray2,
+        Gray4,
+        Gray8,
+        Gray16,
+        Gray32Float,
+        Rgba64
+    }
+
+    public struct PixelFormat : IEquatable<PixelFormat>
```

Comment:
Can be a "recrod struct" without IEquatable/Equals and stuff
https://sharplab.io/#v2:EYLgtghglgdgPgAQEwEYCwAoTACX3YAuApgE4wQA22RMArmNgApQAeRFAYgPYmQECidMDjwBvEXjwAlAObAArADZ5AGgmTss4BAAcenWowa8AIRkld+w8dwmKEAMYBrAOoALKMWs2A4hYCeSN7GfhD+ACzBGqH+BuqSMSiKUQkBAMxIHBRcEAQp0nIQiuHqAL7q6ghp2CREDjwAJtgAzgQktA4ETKzs3Ly56uJGNoSk5FTMbJw8fIL02H2zQgDcmPF4VfgwXSaezYykk+zruEM2GjJEBCcaZ+cjAGbYABSLuXMMALyf3VNvAkIAHR2RyuDzEACUN3uGwA7NgUKthjDJOxmkR8E9XjN3kJsN9fr0cQD6ICYkgociUcYEPCkEjqaiKOjMS9/h98T8jtN+iSwGSAuFKYybLTsOEGSLqMyMVAsey8QTuQrSTEdMKpZIxTpJSK0bL5cSOUqejylqStEp5NhoZq4HAFkbFVzTSr+YlFBrNbgxUldYz9azsbzjS6/k6LYVil7vWLikjbTT4Rl/edylTJOmbqMyJRsMAuFwqAAJCDNACCFAADm4IPiAHyOkPOwlm3GR7T6HQ2jPe6n2pvmr5honN0lmCxdxN9jQDt2c1tuwFaIoSta93A58aL4nPZURhgPYkxyR3GHzn5H3mpvBZjfYAD0D82CCSO95mjkVob774zWXX7KDeuBPi+b77h+K5dj+EF/gBnb6MBj7PtUr6KL+uTYBOlh6DBrrEv+2Fdkh67nJsXAAG6kCQUANBir4AAzYAAKlwADKbSwDIzwQj+S6sRxNEwNxEL+qUQA==

Either way it's still can be created from the user code with public ctor